python-checks/src/main/java/org/sonar/python/checks/hotspots/HashingDataCheck.java

```diff
@@ -0,0 +1,195 @@
+    "ldap_pbkdf2_sha256", "ldap_pbkdf2_sha512", "ldap_plaintext", "ldap_salted_md5",
+    "ldap_salted_sha1", "ldap_sha1", "ldap_sha1_crypt", "ldap_sha256_crypt",
+    "ldap_sha512_crypt", "lmhash", "md5_crypt", "msdcc", "msdcc2",
+    "mssql2000", "mssql2005", "mysql323", "mysql41", "nthash", "oracle10",
+    "oracle11", "pbkdf2_sha1", "pbkdf2_sha256", "pbkdf2_sha512", "phpass", "plaintext",
+    "postgres_md5", "roundup_plaintext", "scram", "scrypt", "sha1_crypt", "sha256_crypt",
+    "sha512_crypt", "sun_md5_crypt", "unix_disabled", "unix_fallback")
+    .stream()
+    .map(hasher -> "passlib.hash." + hasher)
+    .collect(Collectors.toSet());
+
+
+  private static final Set<String> questionableDjangoHashers = immutableSet(
+    "PBKDF2PasswordHasher", "PBKDF2SHA1PasswordHasher", "Argon2PasswordHasher",
+    "BCryptSHA256PasswordHasher", "BasePasswordHasher", "BCryptPasswordHasher", "SHA1PasswordHasher", "MD5PasswordHasher",
+    "UnsaltedSHA1PasswordHasher", "UnsaltedMD5PasswordHasher", "CryptPasswordHasher")
+    .stream()
+    .map(hasher -> "django.contrib.auth.hashers." + hasher)
+    .collect(Collectors.toSet());
+
+
+  @Override
+  public Set<AstNodeType> subscribedKinds() {
+    return immutableSet(
+      PythonGrammar.CALL_EXPR,
+      PythonGrammar.ATTRIBUTE_REF,
+      PythonGrammar.ATOM,
+      PythonGrammar.EXPRESSION_STMT,
+      PythonGrammar.CLASSDEF);
+  }
+
+  @Override
+  public void visitNode(AstNode node) {
+    if (node.is(PythonGrammar.ATTRIBUTE_REF, PythonGrammar.ATOM)) {
+      checkQuestionableHashingAlgorithm(node);
+
+    } else if (node.is(PythonGrammar.EXPRESSION_STMT)) {
+      checkOverwriteDjangoHashers(node);
+
+    } else if (node.is(PythonGrammar.CLASSDEF)) {
+      checkCreatingCustomHasher(node);
+
+    } else if (node.is(PythonGrammar.CALL_EXPR)) {
+      super.visitNode(node);
+    }
+  }
+
+  @Override
+  protected boolean isException(AstNode callExpression) {
+    if (getQualifiedName(callExpression).equals("django.contrib.auth.hashers.make_password")) {
+      AstNode argList = callExpression.getFirstChild(PythonGrammar.ARGLIST);
+      if (argList != null) {
+        return argList.getChildren(PythonGrammar.ARGUMENT).size() == 1;
+      }
+    }
+    return false;
+  }
+
+  private void checkOverwriteDjangoHashers(AstNode expressionStatement) {
+    AstNode lhsExpression = getLHSExpression(expressionStatement);
+    if (lhsExpression == null) {
+      return;
+    }
+
+    // checks for settings.PASSWORD_HASHERS = value
+    AstNode attributeRef = lhsExpression.getFirstDescendant(PythonGrammar.ATTRIBUTE_REF);
+    if (attributeRef != null) {
+      AstNode atom = attributeRef.getFirstChild(PythonGrammar.ATOM);
+      if (atom != null && getQualifiedName(atom).equals("django.conf.settings") &&
+        attributeRef.getLastChild(PythonGrammar.NAME).getTokenValue().equals("PASSWORD_HASHERS")) {
+        addIssue(expressionStatement, MESSAGE);
```

Comment:
don't you want to explicitly `return` here?
It sounds strange to me that we could raise 2 issues in one run (by reaching both line 128 and 135), but if you think it's OK, then fine.

Reply:
yes indeed, I agree. I'll add a `return`